src/services/route-builder.js

```diff
@@ -0,0 +1,359 @@
+'use strict'
+
+const _ = require('lodash')
+const BigNumber = require('bignumber.js')
+const NoRouteFoundError = require('../errors/no-route-found-error')
+const UnacceptableExpiryError = require('../errors/unacceptable-expiry-error')
+const UnacceptableAmountError = require('../errors/unacceptable-amount-error')
+const LedgerNotConnectedError = require('../errors/ledger-not-connected-error')
+const InvalidAmountSpecifiedError = require('../errors/invalid-amount-specified-error')
+const InvalidPacketError = require('../errors/invalid-packet-error')
+const UnreachableError = require('../errors/unreachable-error')
+const InsufficientTimeoutError = require('../errors/insufficient-timeout-error')
+const Accounts = require('./accounts')
+const RoutingTable = require('./routing-table')
+const RateBackend = require('./rate-backend')
+const Quoter = require('./quoter')
+const Config = require('./config')
+const LiquidityCurve = require('../routing/liquidity-curve')
+const log = require('../common/log').create('route-builder')
+const { getShortestUnambiguousPrefix } = require('../lib/utils')
+
+const PROBE_AMOUNT = new BigNumber(10).pow(14) // stays within 15 max digits for BigNumber from Number
+
+function rateToCurve (rate) {
+  rate = new BigNumber(rate)
+
+  // Make sure that neither amount exceeds 15 significant digits.
+  if (rate.gt(1)) {
+    return new LiquidityCurve([ [0, 0], [ PROBE_AMOUNT / rate, PROBE_AMOUNT ] ])
```

Comment:
Shouldn't this be using bignumber.js' math functions?

Reply:
Copied the existing code to make sure the results are exactly the same - I still like that idea so we don't have to update the tests as much. But we'll see once we get to updating the integration tests how it'll play out.